### ISSUE TITLE
Bundle query-title and post-title into one block

### ISSUE THREAD

**Issue by Sandstromer** [WordPress/gutenberg#31717]
## What problem does this address?

When inserting the post-title block outside the loop - e.g. in a header template-part - the title of the first post in displayed when viewing an archive.
This is expected but it would be awesome if there was a "universal" title block that could be used anywhere.

Suppose a user wants to add the title to the header, this can currently be accomplished by creating two headers.
One for use with index/archive/category/tag/author templates, and another header for use with single/page templates.

With a single universal title block, one header template part would only be required, rather than maintaining more than one header.
Of course it may be desirable to have multiple headers with different layouts but maintaining multiple headers where the only difference is the query/post title could be frustrating.

## What is your proposed solution?

Bundle the query-title and post-title blocks into one block, which would detect ~~what title is required~~ whether we're on an archive or single and display the appropriate title.
When viewing an archive, this universal title block would display the archive title, and when viewing single posts would display the post title.

**Comment by gziolo** [comment]
@ntsekouras, that something you should have the best insights.

**Comment by ntsekouras** [comment]
Hey 👋 - I understand your use case but my first thoughts is that we should keep the concerns separated as they are different things. Having a block names `Post Title` ending up printing something else might be confusing as well, no?

For start the `Archive Title` is just a variation of `Query Title` and in the future there will be more variations added like `Search Title` which could add to the complexity and coupling of the different things.

In addition `Post Title` can be used inside a `Query` block that can live inside an `archive` page so we would have to successfully detect when to show what under the same context. I can't think of an easy way to do this at the top of my head for sure 😄 .

To sum up having multiple headers, in case we want the title in template part header, for `single` and `archive` (for now at least) seems like a reasonable compromise to me, contrary to the complexity/coupling of the bundling and taking into account that the template part is something that is usually not changing too often. 

I would love any other insights though.

**Comment by ntsekouras** [comment]
My concerns above about the coupling of the two blocks still stand and the issue had not had any traction for many years. Closing now,  but can revisit if there's renewed interest.
